src/main/java/com/blackduck/integration/detect/lifecycle/run/step/ContainerScanStepRunner.java

```diff
@@ -95,14 +103,31 @@ public Optional<UUID> invokeContainerScanningWorkflow() {
             uploadImageMetadataToStorageService();
             operationRunner.publishContainerSuccess();
             logger.info("Container scan image uploaded successfully.");
-        } catch (IntegrationTimeoutException e) {
+        } catch (IntegrationTimeoutException e) { // TOME: figure out whether this can apply to SCASS or new BDBA
             operationRunner.publishContainerTimeout(e);
             return Optional.empty();
         } catch (IntegrationException | IOException | OperationException e) {
             operationRunner.publishContainerFailure(e);
             return Optional.empty();
         }
-        return Optional.ofNullable(scanId);
+        return Optional.ofNullable(scanId.toString());
+    }
+
+    private String tryScassOrBdba() throws OperationException, IntegrationException {
+        //NameVersion projectNameVersion, File binaryFile, BlackDuckRunData blackDuckRunData, String type, Gson gson
+        ScanCreationResponse scanCreationResponse = operationRunner.initiateScan(projectNameVersion, containerImage, blackDuckRunData, "CONTAINER", gson);
+
+        String scanId = scanCreationResponse.getScanId();
+        String uploadUrl = scanCreationResponse.getUploadUrl();
+
+        if (StringUtils.isNotEmpty(uploadUrl)) {
```

Comment:
`uploadUrl != null && !uploadUrl.isEmpty()` will remove the need for StringUtils.